src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -723,9 +724,18 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
 
       // Create the cluster
       createClusterConfig = CreateClusterConfig(machineConfig, initScript, serviceAccountInfo.clusterServiceAccount, credentialsFileName, stagingBucket, clusterScopes, clusterVPCSettings, clusterRequest.properties)
-      operation <- gdDAO.createCluster(googleProject, clusterName, createClusterConfig)
+      retryResult <- retryExponentially(whenGoogleZoneCapacityIssue, "Cluster creation failed because zone with adequate resources was not found") { () =>
+        gdDAO.createCluster(googleProject, clusterName, createClusterConfig)
+      }
+      operation <- retryResult match {
+        case Right(value) => Future.successful(value._2)
```

Comment:
There are 3 possible cases here:
1. The request succeeds the first time, in which case we'll have a `Right((List.empty, result))`
2. The request gets retried but eventually succeeds. In which we'll have a `Right((errors, result))`.
3. The request gets retried and never succeeds. In which case we'll have a `Left(errors)`.

We should probably increment the counter for cases 2 and 3, but the current code is only incrementing for case 3.

Reply:
Does it matter than that a cluster that fails once with this error and then succeeds has the same impact on the counter as a cluster that gets recreated and fails multiple times with this error?

Reply:
That's a good question. I'm inclined to increment the counter anytime this error occurs, even if it's part of a retry. We should add separate instrumentation for Error'd clusters, broken out by cause.